frontend/controller/serviceworkers/sw-primary.js

```diff
@@ -45,3 +49,32 @@ self.addEventListener('message', function (event) {
     console.error('[sw] unexpected data:', event.data)
   }
 })
+
+self.addEventListener('push', function (event) {
+  // PushEvent reference: https://developer.mozilla.org/en-US/docs/Web/API/PushEvent
+
+  if (!(self.Notification && self.Notification.permission === 'granted')) {
+    return
```

Comment:
Think it would be a good idea to have `console.debug` here just to note that we got a notification but aren't displaying it.